app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/Adapters/FavoritesListAdapter.java

```diff
@@ -70,46 +72,53 @@ public ViewHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
         } else {
             FavoritesListAdapter.TextAdapterViewHolder holder2 = (FavoritesListAdapter.TextAdapterViewHolder)holder;
-            holder2.getFavoriteName().setText(mFavList[position].getStartPlace().getName() + "--" +
-                    mFavList[position].getEndPlace().getName());
-
-            Map<String, String> mapString = new HashMap<String, String>();
-            mapString.put("start", mFavList[position].getStartPlace().toString());
-            mapString.put("end", mFavList[position].getEndPlace().toString());
-            mapString.put("arriveBy", String.valueOf(false));
-            mapString.put("destinationName", mFavList[position].getEndPlace().getName());
+            holder2.getFavoriteName().setText(mFavList.get(position).getStartPlace().getName() + "--" +
+                    mFavList.get(position).getEndPlace().getName());
 
             Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("\"America/NewYork\""));
             int secondsEpoch = (int) (calendar.getTimeInMillis()/1000L);
-            mapString.put("time", String.valueOf(secondsEpoch));
-
-            Endpoint searchEndpoint = new Endpoint()
-                    .queryItems(mapString)
-                    .path("route")
-                    .method(Endpoint.Method.GET);
-
-            FutureNovaRequest.make(Route[].class, searchEndpoint).thenAccept((APIResponse<Route[]> response) -> {
 
-                ArrayList<Route> routes = new ArrayList<Route>();
-                for(Route r:  response.getData()){
-                    routes.add(r);
-                }
-                mAllRoutesToFavorites.put(position, routes);
-                mOptimalRoutes[position] = routes.get(0);
+            // Getting the other route options
+            Map<String, String> map = new HashMap<String, String>();
+            map.put("Content-Type", "application/json");
+            JSONObject searchJSON = new JSONObject();
+            try {
+                searchJSON.put("start", mFavList.get(position).getStartPlace().toString());
+                searchJSON.put("end", mFavList.get(position).getEndPlace().toString());
+                searchJSON.put("destinationName", mFavList.get(position).getEndPlace().getName());
+                searchJSON.put("arriveBy", false);
+                searchJSON.put("time", secondsEpoch);
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+
+            final RequestBody requestBody =
+                    RequestBody.create(MediaType.get("application/json; charset=utf-8"), searchJSON.toString());
+
+            Endpoint searchEndpoint2 = new Endpoint()
+                    .path("v2/route")
+                    .body(Optional.of(requestBody))
+                    .headers(map)
+                    .method(Endpoint.Method.POST);
+
+            FutureNovaRequest.make(SectionedRoutes.class, searchEndpoint2).thenAccept((APIResponse<SectionedRoutes> response) -> {
```

Comment:
This is probably a perfectly fine way to handle network activity, but what exactly happens if we don't get a response back due to shoddy connections? Do we have a way to indicate loading or network failure on the user's end?

Reply:
Do we have designs for network failures?